### ISSUE TITLE
Parsing a 1 character string does not create a TextNode in the Document

### ISSUE THREAD

**Issue by Shusshu** [inversoft/prime-transformer#1]
When I try to parse the following string: "a" it does not end up in a TextNode.

**Comment by Shusshu** [comment]
It's because of this:

`

```
 while(index <= source.length) {
     previous = state;

     if (index == source.length) {
          state = State.complete;
     }
....
}`
```

The state is Text but index is 1 and source.length is 1 so it changes the state to complete and the handleDocumentCleanup does not receive any outstanding TextNode

**Comment by robotdan** [comment]
Hard to believe I didn't have a unit test for a single character. :-) 

Thank you for letting me know and for submitting a pull request, much appreciated. Your suggested patch does work - however it breaks a few other edge cases. I moved the change into the `handleDocumentCleanup`. 

I checked in a fix under commit 1fc1bedcaaf1a05cef995c887d791ed6d9238718.

**Comment by robotdan** [comment]
The unit tests should cover this case now - but I'll wait to close this issue until you've tested out the patch and verified it resolves your problem.

Are you using this library in production or just testing some different options? If you're testing it against other libraries - have you found any major features we're missing that would keep you from wanting to use it?

Thanks, nice to know someone else has found this library useful.

**Comment by Shusshu** [comment]
Yes it works well.

I'm using this lib in an Android app to be soon in production.
I had to convert your lib to use Java 7 only :( to be able to use it.

I only found one other lib but this one was way way better!
It does all I need as we also extends bbcode to use tags like [PHONE]

Thanks for the lib!

**Comment by robotdan** [comment]
Cool, glad it works for you. For the **`[PHONE]`** tag - you can set a tag with an attribute called **`standalone`** which I think will do what you want. We intended it for use for emoji BBCodes - ones that have no body and no closing tag.

I'd be curious to know if the tag attributes we have in place don't fit your needs - and if not, how I could enhance it to make it work out of the box for your **`[PHONE]`** tag.

Thanks for the feedback.

**Comment by Shusshu** [comment]
[PHONE]+325458787[/PHONE] or  [PHONE=+325458787]05458787[/PHONE] or  [PHONE="+325458787"]05458787[/PHONE]  works out of the box.

So if I would like to use a [BR] tag for next line I would need to put [BR standalone] to make it work? nice but also a lot of characters to type ;)

**Comment by robotdan** [comment]
Fix included in released version 2.0.7.
